crates/rattler_installs_packages/src/python_env/venv.rs

```diff
@@ -291,11 +291,28 @@ prompt = {}"#,
                 .expect("cannot get system python parent folder");
             for bin_name in python_bins.into_iter() {
                 let original_python_bin = original_python_bin_dir.join(bin_name);
+                let original_python_scripts = original_python_bin_dir
+                    .join("Lib/venv/scripts/nt")
+                    .join(bin_name);
+                let venv_python_bin = venv_bin.join(bin_name);
 
-                if original_python_bin.exists() {
-                    let venv_python_bin = venv_bin.join(bin_name);
-                    if !venv_python_bin.exists() {
-                        copy_file(venv_exe_path, &venv_python_bin)?;
+                if original_python_bin.exists() && !venv_python_bin.exists() {
+                    if original_python_scripts.exists() {
+                        copy_file(original_python_scripts, &venv_python_bin)?;
+                    } else {
+                        // edge case for when used python is built from source code
+                        // or we use venv from build
+                        let mut launcher_bin_name = bin_name.to_owned();
+
+                        if bin_name.contains("python") {
+                            launcher_bin_name = bin_name.replace("python", "venvlauncher");
+                        } else if bin_name.contains("pythonw") {
+                            launcher_bin_name = bin_name.replace("pythonw", "venvwlauncher");
+                        }
```

Comment:
Usually in these cases I make use of the feature that rust `if`'s are also statements.

```suggestion
                        let launcher_bin_name = bin_name.to_owned();

                        let launcher_bin_name = if bin_name.contains("python") {
                            bin_name.replace("python", "venvlauncher")
                        } else if bin_name.contains("pythonw") {
                            bin_name.replace("pythonw", "venvwlauncher");
                        }
```

Also using the fact that you can use `let` to shadow variables

Reply:
thanks! I've missed this elegant way :(